Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -7676,40 +7676,123 @@ def to_display(self):
 
 class IndicatorsSearcher:
     """Used in order to search indicators by the paging or serachAfter param
-    :type page: ``int``
+    :type page: ``Optional[int]``
     :param page: the number of page from which we start search indicators from.
 
-    :type filter_fields: ``str``
+    :type filter_fields: ``Optional[str]``
     :param filter_fields: comma separated fields to filter (e.g. "value,type")
 
+    :type from_date: ``Optional[str]``
+    :param from_date: the start date to search from.
+
+    :type query: ``Optional[str]``
+    :param query: indicator search query
+
+    :type size: ``int``
+    :param size: limit the number of returned results.
+
+    :type to_date: ``Optional[str]``
+    :param to_date: the end date to search until to.
+
+    :type value: ``str``
+    :param value: the indicator value to search.
+
+    :type limit ``Optional[int]``
+    :param limit the upper limit of the search
+
     :return: No data returned
     :rtype: ``None``
     """
-    def __init__(self, page=0, filter_fields=None):
+    def __init__(self,
+                 page=0,
+                 filter_fields=None,
+                 from_date=None,
+                 query=None,
+                 size=100,
+                 to_date=None,
+                 value='',
+                 limit=None):
         # searchAfter is available in searchIndicators from version 6.1.0
         self._can_use_search_after = is_demisto_version_ge('6.1.0')
         # populateFields merged in https://github.com/demisto/server/pull/18398
         self._can_use_filter_fields = is_demisto_version_ge('6.1.0', build_number='1095800')
         self._search_after_title = 'searchAfter'
         self._search_after_param = None
+        self._original_page = page
         self._page = page
         self._filter_fields = filter_fields
+        self._total = None
+        self._from_date = from_date
+        self._query = query
+        self._size = size
+        self._to_date = to_date
+        self._value = value
+        self._original_limit = limit
+        self._limit = limit
+
+    def __iter__(self):
+        self._total = None
+        self._search_after_param = None
+        self._page = self._original_page
+        self.limit = self._original_limit
+        return self
+
+    def __next__(self):
+        if self._is_search_done():
```

Comment:
Can be moved to the fetch len line.

Reply:
on further check, it cannot, since `search_indicators_by_version` advances inner properties.